iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/task/subtask/processor/PipeProcessorSubtask.java

```diff
@@ -199,6 +225,58 @@ public int hashCode() {
     return taskID.hashCode();
   }
 
+  //////////////////////////// Parsing events ////////////////////////////
+
+  /** Parse an event by pattern and time, then add parsed events into {@code parsedEventQueue} */
+  private void parseEventByPatternAndTime(Event sourceEvent) throws Exception {
```

Comment:
The parsed event's reference count shall be decreased, and the newly generated events' reference count shall be increased, to avoid wal/tsfile resource leak.

Reply:
If a tsFile event generates no tablets, it shall be reported immediately; If not, the report shall be postponed to the sent time of the last raw tablet.